token-swap/program/fuzz/src/instructions.rs

```diff
@@ -0,0 +1,288 @@
+use spl_token_swap_fuzz::{
+    native_account_data::NativeAccountData, native_token::get_token_balance,
+    native_token_swap::NativeTokenSwap,
+};
+
+use spl_token_swap::{
+    curve::{
+        base::{CurveType, SwapCurve},
+        constant_product::ConstantProductCurve,
+    },
+    error::SwapError,
+    instruction::{Deposit, Swap, Withdraw},
+};
+
+use spl_token::error::TokenError;
+
+use honggfuzz::fuzz;
+
+use arbitrary::Arbitrary;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug, Arbitrary, Clone)]
+enum FuzzInstruction {
+    Swap {
+        token_a_id: AccountId,
+        token_b_id: AccountId,
+        trade_direction: TradeDirection,
+        instruction: Swap,
+    },
+    Deposit {
+        token_a_id: AccountId,
+        token_b_id: AccountId,
+        pool_token_id: AccountId,
+        instruction: Deposit,
+    },
+    Withdraw {
+        token_a_id: AccountId,
+        token_b_id: AccountId,
+        pool_token_id: AccountId,
+        instruction: Withdraw,
+    },
+}
+
+/// Helper enum to tell which direction a swap is meant to go.
+#[derive(Debug, Arbitrary, Clone)]
+enum TradeDirection {
+    AtoB,
+    BtoA,
+}
+
+/// Use u8 as an account id to simplify the address space and re-use accounts
+/// more often.
+type AccountId = u8;
+
+const INITIAL_SWAP_TOKEN_A_AMOUNT: u64 = 100_000_000_000;
+const INITIAL_SWAP_TOKEN_B_AMOUNT: u64 = 300_000_000_000;
+
+const INITIAL_USER_TOKEN_A_AMOUNT: u64 = 1_000_000_000;
+const INITIAL_USER_TOKEN_B_AMOUNT: u64 = 3_000_000_000;
+
+fn main() {
+    loop {
+        fuzz!(|fuzz_instructions: Vec<FuzzInstruction>| {
+            run_fuzz_instructions(fuzz_instructions)
+        });
+    }
+}
+
+fn run_fuzz_instructions(fuzz_instructions: Vec<FuzzInstruction>) {
```

Comment:
nit: avoid requiring a `Vec` when a slice will do
```suggestion
fn run_fuzz_instructions(fuzz_instructions: &[FuzzInstruction]) {
```